### DISCUSSION TITLE
API result for canny is much less useful than from webui result.

### DISCUSSION THREAD

**Discussion by ammartaher** [Mikubill/sd-webui-controlnet#2659] [Q&A] [answered]
I am trying to use the api to pass an image to be processed by canny and used in generation.
If i pass the photo through API it results in img 1 
if i pass the same photo in webui it results in img2 .
In Img1 it is clear that the images are usually almost useless, as if the model is not applied or something!! 

Any idea what am i doing wrong? 

<img width="595" alt="image" src="https://github.com/Mikubill/sd-webui-controlnet/assets/39812337/3225c18c-3ba5-46e2-8e93-f0cbe5518263">

<img width="1082" alt="image" src="https://github.com/Mikubill/sd-webui-controlnet/assets/39812337/a42c5324-c992-40af-a694-6c764e960bd9">

I am using the [ultraspice_xlturbo model](https://civitai.com/models/229392?modelVersionId=342732). and the api payload like this:

int stepsLoad = 5;
double cfgscaleLoad = 2;
try
{ 
    var payload = new
    {
        prompt = prompted,
        steps = stepsLoad,
        cfg_scale = cfgscaleLoad,
        sampler_name = "Euler a",
        height = height,
        width = width,
        alwayson_scripts = new
        {
            controlnet = new
            {
                args = new object []
                {
                    new
                    {
                        enabled = true,
                        image = imageControlCoded,
                        module = "canny",
                        model = "diffusers_xl_canny_full [2b69fca4]",
                        lowvram = true,
                        threshold_a = 100,
                        threshold_b= 200,
                        processor_res = 512,
                    },  
                }
            }
        }
    };
    string payloadJson = JsonConvert.SerializeObject(payload);
    StringContent content = new StringContent(payloadJson, Encoding.UTF8, "application/json");
    HttpResponseMessage response = await _httpClient.PostAsync("/sdapi/v1/txt2img", content);

**Comment by huchenlei** [comment, marked as answer]
You are missing control mode param in ControlNet unit. The default is balanced, but in UI you are using prompt is more important. You can use this extension to dump out API payload from WebUI generation: https://github.com/huchenlei/sd-webui-api-payload-display

**Reply by ammartaher** [reply]
That extension is super helpful indeed, i just copied the whole payload and adjusted what i need and kept others i didn't know i needed and eventually the results were similar 👍
